apps/vaccines-updates-client/src/hooks/useSendReport.ts

```diff
@@ -0,0 +1,30 @@
+import { VaccinesReport } from '@vacgaps/interfaces';
+import { useAuthentication, useFormData } from '../providers';
+import { useCallback } from 'react';
+import { sendReport } from '../services/vacgaps-api-client';
+
+export const useSendReport: () => () => Promise<void> = () => {
+    const formData = useFormData();
```

Comment:
i hope i got you right,
because even though the form data provider uses the type with the VacinneReport, the actual data provider type has more properties so the transform to strict VacinneReport instance is a must - to remove the redundant properties, like the "set" methods